packages/amplify-category-geo/src/service-walkthroughs/resourceWalkthrough.ts

```diff
@@ -59,11 +62,45 @@ export async function pricingPlanWalkthrough<T extends ResourceParameters>(
         pricingPlan = PricingPlan.RequestBasedUsage;
     }
     else {
-        const pricingPlanRoutingChoice = await prompter.yesOrNo(
-            'Does your app provide routing or route optimization for commercial assets?',
-            pricingPlan === PricingPlan.MobileAssetManagement ? true : false
-        );
-        pricingPlan = pricingPlanRoutingChoice ? PricingPlan.MobileAssetManagement : PricingPlan.MobileAssetTracking;
+        let mapsSearchUsability = true;
+        if (extendedFlow === true) {
+            mapsSearchUsability = await prompter.yesOrNo(
+                'Does your app need Maps, Location Search or Routing?',
+                pricingPlan !== PricingPlan.RequestBasedUsage
+            )
+        }
+        if (mapsSearchUsability === true) {
+            const pricingPlanRoutingChoice = await prompter.yesOrNo(
+                'Does your app provide routing or route optimization for commercial assets?',
+                pricingPlan === PricingPlan.MobileAssetManagement ? true : false
+            );
+            pricingPlan = pricingPlanRoutingChoice ? PricingPlan.MobileAssetManagement : PricingPlan.MobileAssetTracking;
+        }
+        else {
+            const pricingPlanChoices = [
+                { name: 'Request Based Usage', value: PricingPlan.RequestBasedUsage },
+                { name: 'Mobile Asset Tracking', value: PricingPlan.MobileAssetTracking },
+                { name: 'Mobile Asset Management', value: PricingPlan.MobileAssetManagement },
+                { name: 'Learn More', value: 'LearnMore'}
+            ];
+            const pricingPlanQuestion = "Based on your use case, you may use any of the pricing plans. Select the pricing plan for ALL your Geo resources in the project. We recommend you start with 'Request Based Usage' and then consider one of the other pricing plans as your usage scales";
+
+            let pricingPlanChoice = await prompter.pick<'one', string>(
+                pricingPlanQuestion,
+                pricingPlanChoices,
+                { initial: 0 }
+            );
+            while (pricingPlanChoice === 'LearnMore') {
+                open(apiDocs.pricingPlan, { wait: false });
+                pricingPlanChoice = await prompter.pick<'one', string>(
+                    pricingPlanQuestion,
+                    pricingPlanChoices,
+                    { initial: 0 }
+                );
+            }
```

Comment:
this is a rare opportunity to break out the do-while loop if you want ;)

Reply:
Can I say pass on this one? ;)